assets/js/modules/ads/components/setup/SetupMainPAX.js

```diff
@@ -0,0 +1,234 @@
+import { CORE_LOCATION } from '../../../../googlesitekit/datastore/location/constants';
+import { ADWORDS_SCOPE, MODULES_ADS } from '../../datastore/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import useQueryArg from '../../../../hooks/useQueryArg';
+import { addQueryArgs } from '@wordpress/url';
+import PAXEmbeddedApp from '../PAXEmbeddedApp';
+const { useSelect, useDispatch, useRegistry } = Data;
+
+export default function SetupMainPAX( { finishSetup } ) {
+	const [ showPaxApp, setShowPaxApp ] = useQueryArg( 'pax' );
+
+	const registry = useRegistry();
+
+	const isAdBlockerActive = useSelect( ( select ) =>
+		select( CORE_USER ).isAdBlockerActive()
+	);
+	const hasAdwordsScope = useSelect( ( select ) =>
+		select( CORE_USER ).hasScope( ADWORDS_SCOPE )
+	);
+
+	const redirectURL = addQueryArgs( global.location.href, {
+		pax: 1,
+		notification: 'ads', // Arbitrary value to prevent authentication_success from surfacing.
+	} );
+
+	const oAuthURL = useSelect( ( select ) =>
+		select( CORE_USER ).getConnectURL( {
+			additionalScopes: [ ADWORDS_SCOPE ],
+			redirectURL,
+		} )
+	);
+	const isNavigatingToOAuthURL = useSelect( ( select ) => {
+		if ( ! oAuthURL ) {
+			return false;
+		}
+
+		return select( CORE_LOCATION ).isNavigatingTo( oAuthURL );
+	} );
+
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+	const { setPaxConversionID, setExtCustomerID, submitChanges } =
+		useDispatch( MODULES_ADS );
+
+	const onLaunch = useCallback(
+		async ( app ) => {
+			if ( app && typeof app?.getServices === 'function' ) {
+				/* eslint-disable sitekit/acronym-case */
+				// Disabling rule because function and property names
+				// are expected in current format by PAX API.
+				const customerData = await app
+					.getServices()
+					?.accountService?.getAccountId( {} );
+				const conversionTrackingData = await app
+					.getServices()
+					?.conversionTrackingIdService?.getConversionTrackingId(
+						{}
+					);
+
+				if (
+					customerData.externalCustomerId &&
+					conversionTrackingData.conversionTrackingId
+				) {
+					setExtCustomerID( customerData.externalCustomerId );
+					setPaxConversionID(
+						conversionTrackingData.conversionTrackingId
+					);
+					/* eslint-enable sitekit/acronym-case */
+
+					const { error } = await submitChanges();
+
+					if ( error ) {
+						return;
+					}
+
+					// This callback is often invoked before all site info is resolved.
+					// We need to ensure data is resolved, otherwise the admin URL will be
+					// `undefined` and `finishSetup` will error instead of redirecting
+					// to the dashboard.
+					await registry
+						.__experimentalResolveSelect( CORE_SITE )
+						.getSiteInfo();
+
+					const adminURL = registry
+						.select( CORE_SITE )
+						.getAdminURL( 'googlesitekit-dashboard', {
+							notification: 'authentication_success',
+							slug: 'ads',
+						} );
+
+					finishSetup( adminURL );
```

Comment:
+1 for using the arg to this function, however I don't think this would be necessary until #8661 it seems the default has been implemented here.

Reply:
The problem is that this can be invoked pretty early before the default data is resolved, resulting in console error, and no redirect. Need to keep this as it ensures this works every time

Reply:
It shouldn't be invoked until the user clicks "Complete setup" so it shouldn't be possible to call this too early since the button should be disabled until it is ready.